### ISSUE TITLE
Use a command-line parsing library

### ISSUE THREAD

**Issue by watkipet** [hoche/splat#33]
The present command-line parsing is difficult to understand and brittle. It's also subject to all sorts of buffer-overrun attacks (bad news for people putting a web interface in front of Splat! who don't sanitize their input). I suggest using a the 3rd party library for parsing command line arguments. The ones that seem reasonable (in order of what I think I'd prefer) are:

1. CLI11
2. gflags
3. tclap

Any concerns about using the 3rd party library? Any preference for one of these (or another)?

**Comment by der-stefan** [comment]
Good point stating the possible attack vulerability. Again, as in #32 a short list of dependencies is desireable. Let's find a good compromise. 👍

**Comment by dBitech** [comment]
I have not used any of these libs in particular but from an available documentation/feature set point of view, I think CLI11 is a good fit.   I had also looked at argparse, but CI11 seemed to be a better fit

**Comment by watkipet** [comment]
OK I'll investigate using CL11. CL11 is header-only. Unlike some other (even header-only) libraries, it's not in MacPorts, Ubuntu, or Debian's main package repos. I'll investigate how it's usually integrated with CMake.

**Comment by hoche** [comment]
That branch got out of date so I redid it from scratch. I also updated the docs.
It's a breaking change because of all the changes from single-dash to double-dash.

It will be in 2.0.1.
